crates/pathfinder/src/sync/class_definitions.rs

```diff
@@ -222,11 +235,16 @@ fn compute_hash_impl(input: ClassWithLayout) -> anyhow::Result<Class> {
         ),
     }?;
 
-    Ok(Class {
-        block_number,
-        definition,
-        hash,
-    })
+    if hash != layout.hash {
+        tracing::debug!(input_hash=%layout.hash, actual_hash=%hash, "Class hash mismatch");
+        Err(SyncError2::ClassHashComputationError.into())
```

Comment:
nit: I wonder if we should differentiate between a mismatch (ie. proper layout, bad values) and hash computation error (which means bad layout).
So for the former: ClassHashMismatch (maybe we need to add this variant)
and for the latter: ClassHashComputationError (which really is an InvalidDto but at a higher level)

Reply:
Well, I wasn't keen on extending SyncError / SyncError2, b/c